_posts/2015-02-20-gocumber.markdown

```diff
@@ -0,0 +1,91 @@
+
+1. We write Gherkin features as a group to lay out our desired behavior. This is done before any code is written or even a git repository is initialized.
+1. We then set up our project and pull in all of our desired libraries
+1. Next we begin implementing the Gherkin steps until we can run our features (with failures since nothing is implemented)
+1. Lastly we start implementing in code
+
+The last step there also involves unit testing but I'll leave that out for the time being. Rest assured no code is written until we have a lot of failing tests.
+
+## Okay, so you like to test. Now what?
+All of this meant that the first order of business after our Gherkin features were written was picking our testing tools. The go standard library has a pretty robust [testing](http://golang.org/pkg/testing/) package that seemed to offer what we needed but we quickly realized that there was nothing out there that provided what we needed for our Gherkin features.
+
+Currently at Sittercity we mostly have applications written in Ruby. This means that most of our other projects use [cucumber](https://cukes.info/). But since we are expanding into Go we needed some way to parse and execute our Gherkin features. We found a few options but they were either not actively developed or half-implemented. We decided to take a first stab at implementing what we needed ourselves and see where it took us.
+
+For starters we needed some way to parse the Gherkin itself. Luckily we found the [go-gherkin](https://github.com/muhqu/go-gherkin) package from [muhqu](https://github.com/muhqu). This allowed us to concentrate on creating an interface to execute steps and access variables.
+
+## Gocumber Example
+So what does this interface look like in code? Let's start with some simple gherkin:
+
+{% highlight gherkin %}
+Feature: gocumber example feature
+  As a person wanting to learn about gocumber
+  I want to be shown a bunch of awesome examples
+
+  Scenario: Deactivate a user
+    Given I have an existing user with the name "Cool Dude"
+    When I deactivate the user named "Cool Dude"
+    Then the user "Cool Dude" should be deactivated
+{% endhighlight %}
+
+Now we need to implement the steps listed above. We would create a file named 'example_feature_test.go' and set it up like a standard Go test:
+
+{% highlight go %}
+import "testing"
+import "github.com/sittercity/gocumber"
+
+func TestYourStuff(t *testing.T) {
+  steps := make(gocumber.Definitions)
+
+  steps.Given(`I have an existing user with name "(.*)"`, func(matches []string, _ gocumber.StepNode) {
+    // your code to create a user, you can use 'matches[1]' to pull out the
+    // name, like so: user_name := matches[1]
+  })
+
+  steps.When(`I deactivate the user named "(.*)"`, func(matches []string, _gocumber.StepNode) {
```

Comment:
Addressed in https://github.com/sittercity/techblog/commit/d178bc4c14000652f3970f5f98673ce0cc59b49a